app_base.html

```diff
@@ -0,0 +1,47 @@
+<%inherit file = "/homepage/templates/app_base.htm" />
+<%! from catalog import models as cmod%>
+
+
+<%block name ="left_column">
+
+  <div id ='active_section'>
+      <ul id="category_list">
+
+      <li>
+          <a class = '${'activeone' if cat_name == 'All Products' else ''}' href="/catalog/index/0/">All Products</a>
+      </li>
+
+        % for categories in cmod.Category.objects.all():
+            <li>
+                <a class ='${ 'activeone' if cat_name == categories.name else ''}' href="/catalog/index/${ categories.id}/">${categories.name}</a>
+            </li>
+        % endfor
+      </ul>
+  </div>
+</%block>
```

Comment:
spacing between lines is a little inconsistent; can make it unnecessarily harder to understand the code

Reply:
Perfectionists..